src/Runtime/OMTensorList.inc

```diff
@@ -103,23 +116,18 @@ void omTensorListDestroy(OMTensorList *list) {
     return;
   for (int64_t i = 0; i < list->_size; i++)
     omTensorDestroy(list->_omts[i]);
-  if (list->_owning) {
-    free(list->_omts);
-  }
+  // Free the list as well as the pointers to the OMTensor array
```

Comment:
For the destroy, should we check if the shallow is set?

Reply:
This get's kind of interesting. What if you do OMTensorListCreateShallow(true) and then call OMTensorListDestroy?

For backwards compatibility we've implied OMTensorListDestroy means deep since that was the original behavior. 

We could say this is controlled at List creation time instead, but then what happens if you do `OMTensorListCreateShallow(false)` and then call `OMTensorListDestroyShallow` later. Which behavior do you honor, the one requested at creation or the one requested at destruction?

TBH I don't have a great answer. My review suggestion was to document that `_shallow` only applies when `delete` is used instead of one of the explicit Destroy methods.

Reply:
@gongsu832 We had the same exact precedent for `omTensorListCreateWithOwnership` where we had two constructors one for a user explicitly specifying true/false and the second for the case where a user does not specify anything which is defaulted to false. We had no issue with carrying out this implementation before and essentially it is similar logic....just the names of the API and the flag itself have changed. But the overall structure is in sync with what we had previously. 

However, if we only want `OMTensorListCreate` .....I need to know for sure before I go on vacation. To me I do not see an issue to keep things as is but I am flexible with what the majority wants.

Reply:
`omTensorListCreateWithOwnership` was for the `_owning` flag, which is needed from users to tell whether they want to keep the tensor list or not. Controlling deep/shallow destroy, on the other hand, does NOT require a flag (the C++ `_shallow` is due to the way how C++ construct/destructor works). So they are two entirely different matters. Adding `omTensorListCreateShallow` adds nothing but additional complications when users call mismatched create and destroy. So just because we used one implementation in one doesn't mean that it makes sense in the other.

Reply:
@gongsu832 We had the same exact precedent for `omTensorListCreateWithOwnership` where we had two constructors one for a user explicitly specifying true/false and the second for the case where a user does not specify anything which is defaulted to false. We had no issue with carrying out this implementation before and essentially it is similar logic....just the names of the API and the flag itself have changed. But the overall structure is in sync with what we had previously. 

However, if we only have one constructor I guess we can keep it as

Reply:
It does require a flag just like owning.... the shallow flag is required or else the destructor could not decipher between a shallow destroy or a regular destroy. So I do not agree with that statement. We have one destructor and the logic (even from what you updated) is written in away to give indication as to which method the caller will utilize upon destruction. I am fine with removing omTensorListCreateShallow as of now.

Reply:
So what I _think_ @gongsu832 is saying is that for C there should just be:
```
OMTensorList *omTensorListCreate(OMTensor **tensors, int64_t n){}
```
And no `shallow` param. The shallow vs deep destroy is solely controlled at destruction time by whether `omTensorListDestroy` or `omTensorListDestroyShallow` is called.

I'm a little confused on the mechanics for C++ though. Does it need two constructors? Would as it is now, an optional param in the constructor not work?
```
  OMTensorList(OMTensor *omts[], int64_t n, bool shallow = false)?
```

However for C++ that means the internal `_shallow` will need to exist right? Since C++ can call C methods, there's no way to avoid the miss match scenario like:
```
OMTensorList * list = new OMTensorList(arr, 3, true) // choosing deep destory at construct time)
... // use the object
OMTensorListDestoryShallow(list) // choosing shallow at destruction
```
or does something in C++ prevent that and force you to do `delete list` instead?

Reply:
I will remove it....as long as it is done. I am fine with anything

Reply:
> So what I _think_ @gongsu832 is saying is that for C there should just be:
> 
> ```
> OMTensorList *omTensorListCreate(OMTensor **tensors, int64_t n){}
> ```
> 
> And no `shallow` param. The shallow vs deep destroy is solely controlled at destruction time by whether `omTensorListDestroy` or `omTensorListDestroyShallow` is called.

Yes. Because we didn't want to change the existing `omTensorListCreate` and that was the whole reason `omTensorListDestroyShallow` was introduced.


> I'm a little confused on the mechanics for C++ though. Does it need two constructors? Would as it is now, an optional param in the constructor not work?
> 
> ```
>   OMTensorList(OMTensor *omts[], int64_t n, bool shallow = false)?
> ```

Yes the internal `_shallow` will exist, for C++ only. I don't know C++ enough whether a constructor can take an optional parameter or not. I doubt it. But having two constructors or one constructor with an optional parameter will make no difference externally.


> OMTensorList * list = new OMTensorList(arr, 3, true) // choosing deep destory at construct time)
> ... // use the object
> OMTensorListDestoryShallow(list) // choosing shallow at destruction

This is just suicidal code. You are creating a C++ object and passing it to C code. But in any case, we currently don't expose the C++ constructor/destructor API so you can't really create an `OMTensorList` object with `new`.